README.md

```diff
@@ -187,74 +192,137 @@ tx_response = send_reliable_submission(my_tx_payment_signed, client)
->>> tx_response = send_reliable_submission(my_tx_payment_signed, client)
->>> my_tx_payment
-Payment(
-    account='rMPUKmzmDWEX1tQhzQ8oGFNfAEhnWNFwz',
-    transaction_type=<TransactionType.PAYMENT: 'Payment'>,
-    fee=10000,
-    sequence=16034065,
-    account_txn_id=None,
-    flags=0,
-    last_ledger_sequence=10268600,
-    memos=None,
-    signers=None,
-    source_tag=None,
-    signing_pub_key=None,
-    txn_signature=None,
-    amount='2200000',
-    destination='rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe',
-    destination_tag=None,
-    invoice_id=None,
-    paths=None,
-    send_max=None,
-    deliver_min=None
+The `xrpl-py` library automatically populates the `fee`, `sequence` and `last_ledger_sequence` fields when you create transactions. In the example above, you could omit those fields and let the library fill them in for you.
+
+```py
+from xrpl.models.transactions import Payment
+from xrpl.transaction import send_reliable_submission, safe_sign_and_autofill_transaction
+# prepare the transaction
+# the amount is expressed in drops, not XRP
+# see https://xrpl.org/basic-data-types.html#specifying-currency-amounts
+my_tx_payment = Payment(
+    account=test_wallet.classic_address,
+    amount="2200000",
+    destination="rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe"
 )
+
+# sign the transaction with the autofill method
+# (this will auto-populate the fee, sequence, and last_ledger_sequence)
+my_tx_payment_signed = safe_sign_and_autofill_transaction(my_tx_payment, test_wallet, client)
+print(my_tx_payment_signed)
+# Payment(
+#     account='rMPUKmzmDWEX1tQhzQ8oGFNfAEhnWNFwz',
+#     transaction_type=<TransactionType.PAYMENT: 'Payment'>,
+#     fee='10',
+#     sequence=16034065,
+#     account_txn_id=None,
+#     flags=0,
+#     last_ledger_sequence=10268600,
+#     memos=None,
+#     signers=None,
+#     source_tag=None,
+#     signing_pub_key='EDD9540FA398915F0BCBD6E65579C03BE5424836CB68B7EB1D6573F2382156B444',
+#     txn_signature='938FB22AE7FE76CF26FD11F8F97668E175DFAABD2977BCA397233117E7E1C4A1E39681091CC4D6DF21403682803AB54CC21DC4FA2F6848811DEE10FFEF74D809',
+#     amount='2200000',
+#     destination='rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe',
+#     destination_tag=None,
+#     invoice_id=None,
+#     paths=None,
+#     send_max=None,
+#     deliver_min=None
+# )
+
+# submit the transaction
+tx_response = send_reliable_submission(my_tx_payment_signed, client)
 ```
 
 
+### Subscribe to ledger updates
+
+You can send `subscribe` and `unsubscribe` requests only using the WebSocket network client. These request methods allow you to be alerted of certain situations as they occur, such as when a new ledger is declared.
+
+```py
+from xrpl.clients import WebsocketClient
+url = "wss://s.altnet.rippletest.net/"
+client = WebsocketClient(url)
+from xrpl.models.requests import Subscribe, StreamParameter
+req = Subscribe(streams=[StreamParameter.LEDGER])
+client.open()
+client.send(req)
+for message in client:
+    print(message)
```

Comment:
please change to:

```py
from xrpl.clients import WebsocketClient
from xrpl.models.requests import Subscribe, StreamParameter

with WebsocketClient("wss://s.altnet.rippletest.net/") as client:
  client.send(Subscribe(streams=[StreamParameter.LEDGER])
  for message in client:
    print(message)
```

Reply:
we should perhaps note that doing it this way, _without_ a timeout on the sync client, will cause the above code to run forever